src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -614,6 +622,9 @@ public bool Remove(TKey key)
                         {
                             entries[last].next = entries[i].next;
                         }
+
+                        value = entries[i].value;
```

Comment:
This implementation of `public bool Remove(TKey key, out TValue value)` penalizes many users of `public bool Remove(TKey key)` by introducing an unnecessary copy (and initialization) of the value.